receiver/dockerstatsreceiver/metric_helper.go

```diff
@@ -71,3 +78,57 @@ func calculateMemoryPercent(limit uint64, usedNoCache uint64) float64 {
 	}
 	return 0.0
 }
+
+// calculateCPULimit calculate the number of cpus assigned to a container.
+//
+// Calculation is based on 3 alternatives by the following order:
+// - nanocpus:   if set by i.e docker run -cpus=2
+// - cpusetCpus: if set by i.e docker run -docker run -cpuset-cpus="0,2"
+// - cpuquota:   if set by i.e docker run -cpu-quota=50000
+//
+// See https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler for background.
+func calculateCPULimit(hostConfig *ctypes.HostConfig) (float64, error) {
+	var cpuLimit float64
+	var err error
+
+	switch {
+	case hostConfig.NanoCPUs > 0:
+		cpuLimit = float64(hostConfig.NanoCPUs) / nanosInASecond
+	case hostConfig.CpusetCpus != "":
+		cpuLimit, err = parseCPUSet(hostConfig.CpusetCpus)
+		if err != nil {
+			return cpuLimit, err
+		}
+	case hostConfig.CPUQuota > 0:
+		period := hostConfig.CPUPeriod
+		if period == 0 {
+			period = 100000 // Default CFS Period
+		}
```

Comment:
Does a `-1` default make sense opposed to a zero value?

Reply:
I'd say `10000` makes sense here, as it is the [default value](https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler), we'll get `0` (golang's default) here `hostConfig.CPUPeriod` is not set.